### DISCUSSION TITLE
Adjust selected vertical in query string parameter when another vertical is selected.

### DISCUSSION THREAD

**Discussion by VerlindenEAM** [microsoft-search/pnp-modern-search#1911] [Ideas]
**Version used**
4.5.2

**Describe the bug**
If you click on another vertical, the parameter in the url is not adjusted.

**To Reproduce**
1. Go to 'Verticals webpart configuration'
2. Enter 'v' in the configuation of "Query string parameter to use to select a vertical tab by default"
3. Go to Vertical Sites
4. Go to Vertical People

For example:
All | Sites | People | 
If you go to the URL with ?q=<keyword>&v=Sites it will directly show the results from the Sites vertical
If you click on People, it will show the results from the People vertical, but the URL is not adjusted to ?q=<keyword>&v=People

**Expected behavior**
If you change vertical, the parameter should be adjusted in the url.

**Screenshots**

**Desktop (please complete the following information):**
 - Edge

**Additional context**

**Comment by wobba** [comment]
What's the use case, outside of copying the link and sharing with someone?

**Comment by FranckyC** [comment]
This is the expected behavior as the query string parameter is only used to select default tab. I'll move it to discussions.

**Comment by VerlindenEAM** [comment]
The usecase you are suggesting is already a major usecase imo.
Also after page refresh the vertical selection is removed and it will go back to the first vertical (Sometimes a page refresh is required due to SharePoint or webpart issues, for example not loading correctly).
This functionality was part of v3 before, where a parameter to pick up from the query string to default activate a vertical. We are using it in our v3 portals that we build.
If you use verticals with hyperlink to a new page with &v= in it, then the parameter is copied to the new page.
Using v= in the url is already working if you have other sites or pages where you want to directly link to vertical.
For end users it is confusing if the vertical selection is removed after refresh or when you use some verticals pointing to the same page where it is removed and verticals pointing to a new page where the vertical selection will work.
